lib/Sub/WrapInType.pm

```diff
@@ -12,18 +12,20 @@ use Type::Params qw( multisig compile compile_named Invocant );
 use namespace::autoclean;
 
 our $VERSION = '0.04';
-our @EXPORT  = qw( wrap_sub );
+our @EXPORT  = qw( wrap_sub wrap_method );
 
-readonly params  => my %params;
-readonly returns => my %returns;
-readonly code    => my %code;
+readonly params    => my %params;
+readonly returns   => my %returns;
+readonly code      => my %code;
+readonly is_method => my %is_method;
 
 my $TypeConstraint = HasMethods[qw( assert_valid )];
 my $ParamsTypes    = $TypeConstraint | ArrayRef[$TypeConstraint] | Map[Str, $TypeConstraint];
 my $ReturnTypes    = $TypeConstraint | ArrayRef[$TypeConstraint];
 
 sub new {
   my $class = shift;
+  my $opts = shift;
```

Comment:
I would like you to validate the type of $opts using Type::Params.

Reply:
Thanks for the code review!

Reply:
@ybrliiu Fixed it!

Reply:
Thanks.